### ISSUE TITLE
No Automated Backups Work

### ISSUE THREAD

**Issue by lordexorr** [PowerShell/Win32-OpenSSH#2306]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the latest version
- [X] Search the existing issues.

### Steps to reproduce

We are looking to replace the integrated OpenSSH Server in Windows with the Github release so we can better control versioning and patching. We utilize OpenSSH Server for automated (nightly) configuration backups of numerous devices (Networking, VMware, etc). After installing OpenSSH Server v9.8.1.0p1 from Github (version doesn't seem to matter though, we've tried multiple) any manual backup works as expected; but all our automated/nightly backups fail. If we revert back to the built-in Windows OpenSSH Server they immediately start working again. I'm not clear on what the differences are, but I know our sshd_config files are identical, as well as the .pub and .key files. Permissions have been reset using the FixHostFilePermissions.ps1 and FixUserFilePermissions.ps1 scripts. Does anyone have any thoughts on what could be causing this. After MS broke OpenSSH in both October and November patch cycles we are really hoping to get away from utilizing the built-in version to better control this.

### Expected behavior

```console
Automated backups to function normally.
```

### Actual behavior

```console
Automated backups fail.
```

### Error details

```console
I am not able to see any errors that would point to the cause.
```

### Environment data

```PowerShell
Currently using v9.8.1.0p1 but we have tried previous versions, all have the same issue.
```

### Version

v9.8.1.0p1

**Comment by jborean93** [comment]
You'll most likely have to share more details like:

+ How is the backup software interacting with SSH
+ Is ssh acting as the client or server
+ Is ssh even working, what commands are you running

The more information you can share the better someone will be able to help figure out what might be going wrong.

**Comment by mgkuhn** [comment]
In the absence of pretty much any further details that would allow anyone to reproduce your situation, I can only suggest some control experiments to narrow down possible causes. For example, try to install from Github the same older release (9.5?) that Microsoft currently ships with your server, as a control experiment to learn whether the cause of your malfunction is related to either (a) the newer OpenSSH version number 9.8.1, or (b) the way you've installed OpenSSH for Windows from the Github release.

**Comment by mgkuhn** [comment]
> After MS broke OpenSSH in both October and November patch cycles

What exactly broke?

**Comment by lordexorr** [comment]
Thanks to those that responded. After much troubleshooting and comparing configurations, we realized that after the upgrade to OpenSSH Server 9.5 (or later) if we had HostKey configs uncommented the automated backups would fail. When I commented those out of the config, backups would work successfully. Seems to be something related to OpenSSH Server 9.5 and later on Windows Servers. The Oct/Nov Windows Cumulative updates included an OpenSSH Server upgrade from 7.7 to 9.5 which would explain it. MS is aware of the issue and hopefully will release a fix in a future update. The link below has some details on it.

https://learn.microsoft.com/en-us/answers/questions/2115772/sftp-connectivty-issue-reported-with-regard-to-kb5

**Comment by tgauth** [comment]
> Thanks to those that responded. After much troubleshooting and comparing configurations, we realized that after the upgrade to OpenSSH Server 9.5 (or later) if we had HostKey configs uncommented the automated backups would fail. When I commented those out of the config, backups would work successfully. Seems to be something related to OpenSSH Server 9.5 and later on Windows Servers. The Oct/Nov Windows Cumulative updates included an OpenSSH Server upgrade from 7.7 to 9.5 which would explain it. MS is aware of the issue and hopefully will release a fix in a future update. The link below has some details on it.
> 
> https://learn.microsoft.com/en-us/answers/questions/2115772/sftp-connectivty-issue-reported-with-regard-to-kb5

Could you share what the uncommented HostKey lines were? Would the ssh connection fail before the connection was established (pre auth) or during the backup (post auth)?
I'd like to understand if it's related to using a deprecated hostkey type like ssh-rsa or something else.

**Comment by lordexorr** [comment]
Authentication would succeed, but then I would get an error in the OpenSSH Operational logs saying "no matching host key type found" when the backup would try to save data to the SFTP server.

Below are the host keys we commented out.

#HostKey __PROGRAMDATA__/ssh/ssh_host_rsa_key
#HostKey __PROGRAMDATA__/ssh/ssh_host_dsa_key
#HostKey __PROGRAMDATA__/ssh/ssh_host_ecdsa_key
#HostKey __PROGRAMDATA__/ssh/ssh_host_ed25519_key
#HostKey __PROGRAMDATA__/ssh/ecdsa_key
